--- origsrc/gamin-0.1.10/server/gam_fs.c	2009-03-22 13:50:20.656250000 -0500
+++ src/gamin-0.1.10/server/gam_fs.c	2009-02-16 03:09:08.203125000 -0600
@@ -119,7 +119,7 @@ gam_fs_filesystem_sort_cb (gconstpointer
 	return strlen(fsb->path) - strlen (fsa->path);
 }
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__CYGWIN__)
 static void
 gam_fs_scan_mtab (void)
 {
@@ -209,7 +209,7 @@ gam_fs_getmntinfo (void)
 void
 gam_fs_init (void)
 {
-#if defined(__linux__)
+#if defined(__linux__) || defined(__CYGWIN__)
 	if (initialized == FALSE)
 	{
 		initialized = TRUE;
@@ -219,6 +219,8 @@ gam_fs_init (void)
 		gam_fs_set ("reiserfs", GFS_MT_DEFAULT, 0);
 		gam_fs_set ("novfs", GFS_MT_POLL, 30);
 		gam_fs_set ("nfs", GFS_MT_POLL, 5);
+		gam_fs_set ("ntfs", GFS_MT_POLL, 5);
+		gam_fs_set ("vfat", GFS_MT_POLL, 5);
 		if (stat("/etc/mtab", &mtab_sbuf) != 0)
 		{
 			GAM_DEBUG(DEBUG_INFO, "Could not stat /etc/mtab\n");
