--- origsrc/gamin-0.1.10/libgamin/Makefile.am	2008-11-24 11:38:36.271871800 -0600
+++ src/gamin-0.1.10/libgamin/Makefile.am	2008-11-24 11:36:44.000000000 -0600
@@ -40,11 +40,13 @@
 libgamin_1_la_LIBADD =
 
 libgamin_1_la_LDFLAGS = -Wl,--version-script=$(srcdir)/gamin_sym.version \
+                        -no-undefined \
                         -version-info @GAMIN_VERSION_INFO@ @THREAD_LIBS@
 
 libfam_la_SOURCES = $(libgamin_1_la_SOURCES)
 libfam_la_LIBADD = $(libgamin_1_la_LIBADD)
 libfam_la_LDFLAGS = -Wl,--version-script=$(srcdir)/gamin_sym.version	\
+                    -no-undefined \
                     -version-info @FAM_VERSION_INFO@ @THREAD_LIBS@
 
 #
--- origsrc/gamin-0.1.10/python/Makefile.am	2008-11-12 14:45:28.000000000 -0600
+++ src/gamin-0.1.10/python/Makefile.am	2008-11-24 11:37:49.974996800 -0600
@@ -10,8 +10,8 @@
 python_LTLIBRARIES = _gamin.la
 
 _gamin_la_SOURCES = gamin.c
-_gamin_la_LIBADD = $(top_builddir)/libgamin/libgamin-1.la
-_gamin_la_LDFLAGS = -module -avoid-version
+_gamin_la_LIBADD = $(top_builddir)/libgamin/libgamin-1.la -lpython$(PYTHON_VERSION)
+_gamin_la_LDFLAGS = -module -avoid-version -no-undefined -L$(libdir)/python$(PYTHON_VERSION)/config
 
 python_PYTHON = gamin.py
 
